### PR TITLE
Doumentation improvements and fixes

### PR DESCRIPTION
**Pull request by corranwebster**
A general set of improvements to the documentation.  This adds documentation for a few omitted editors, the UI object, and imporives the description of the UIInfo object, plus assorted minor fixes and improvements.

### CODE REVIEW

**Comment by mdickinson** [comment]
Looks like Travis CI notifications are dodgy. Not sure whether this is random flakiness on the part of GitHub / Travis CI or whether there's some configuration that needs fixing.

File: docs/source/traitsui_user_manual/factories_basic.rst [hunk 1]
```diff
@@ -45,10 +45,13 @@ ArrayEditor()
     *width*
 
 The editors generated by ArrayEditor() provide text fields (or static text for
-the read-only style) for each cell of a two-dimensional Numeric array. Only the
+the read-only style) for each cell of a two-dimensional Numpy array. Only the
```
Comment (mdickinson) [inline docs/source/traitsui_user_manual/factories_basic.rst:48]: Spelling nitpick: NumPy, not Numpy.

File: docs/source/traitsui_user_manual/factories_basic.rst [hunk 2]
```diff
@@ -395,23 +398,50 @@ See Also
...
-    (any)
+    Date, List(Date) (custom style only)
 :Default for:
-    (none)
+    Date
+:Optional parameters:
+    *allow_future*, *message*, *months*, *multi_select*, *on_mixed_select*,
+    *padding*, *shift_to_select*, *selected_style*, *strftime*, *view*
+
+The DateEditor() displays a Python datetime date object, usually supplied via a
```
Comment (mdickinson) [inline docs/source/traitsui_user_manual/factories_basic.rst:413]: "datetime date" is slightly confusing; represent as `datetime.date` instead?

File: docs/source/traitsui_user_manual/factories_basic.rst [hunk 3]
```diff
@@ -395,23 +398,50 @@ See Also
...
+Date trait.  The simple style shows a date spin control, while the custom style
+shows one (or potentially more in the wx backend) months in a calendar view.
+Dates can be restricted to past-only by setting *allow_future* to False.
```
Comment (mdickinson) [inline docs/source/traitsui_user_manual/factories_basic.rst:416]: I had no idea that `allow_future` was an option. Given that it is, though, it seems strange that `allow_past` is not. But that's off-topic for this PR.

File: docs/source/traitsui_user_manual/factories_basic.rst [hunk 4]
```diff
@@ -395,23 +398,50 @@ See Also
...
+:Optional parameters:
+    *maximum_datetime*, *message*, *minimum_datetime*, *strftime*
 
-The DefaultOverride() is a factory that takes the trait's default editor and
-customizes it with the specified parameters. This is useful when a trait defines
-a default editor using some of its data, e.g. Range or Enum, and you want to
-tweak some of the other parameters without having recreate that data.
+The DatetimeEditor() displays a Python datetime object, usually supplied
+via a Datetime trait.  The simple style shows a datetime spin control, and maximum
+and minimum selectable datetimess can be supplied to restrict the range of values.
```
Comment (mdickinson) [inline docs/source/traitsui_user_manual/factories_basic.rst:439]: datetimessss, preciousss ...

Comment (mdickinson) [reply docs/source/traitsui_user_manual/factories_basic.rst:439]: Sorry. Ahem. I don't know what came over me there; let me try again. I believe there may be a typo in this line. Please consider replacing the word "datetimess" with "datetimes".

File: docs/source/traitsui_user_manual/factories_basic.rst [hunk 5]
```diff
@@ -837,8 +905,8 @@ Example 16: Instance editor with instance selection
         import View, Item, InstanceEditor
 
     class Person ( HasStrictTraits ):
-        name = Str()
-        age = Int()
+        name  = Str
```
Comment (mdickinson) [inline docs/source/traitsui_user_manual/factories_basic.rst:908]: Urgh. Do we _really_ want to do this?

File: docs/source/traitsui_user_manual/factories_basic.rst [hunk 6]
```diff
@@ -1177,6 +1267,23 @@ You can specify a mapping from user input values to other values with the
...
+````````````
+
+:Suitable for:
+    Time
+:Default for:
+    Time
+:Optional parameters:
+    *minimum_datetime*, *strftime*
+
+The TimeEditor() displays a Python datetime time object, usually supplied
```
Comment (mdickinson) [inline docs/source/traitsui_user_manual/factories_basic.rst:1281]: Same comment as for "datetime date"; I think this is potentially confusing unless it's clear that "datetime" refers to the datetime module.

File: docs/source/traitsui_user_manual/factories_basic.rst [hunk 7]
```diff
@@ -1177,6 +1267,23 @@ You can specify a mapping from user input values to other values with the
...
+via a Time trait.  The simple style shows a time spin control.  For readonly
+style, the user can set the message to display if the datetime value is None,
```
Comment (mdickinson) [inline docs/source/traitsui_user_manual/factories_basic.rst:1283]: Should "datetime value" be "time value" here?

File: docs/source/traitsui_user_manual/handler.rst [hunk 8]
```diff
@@ -45,21 +45,121 @@ means of the UIInfo object.
 
 Whenever TraitsUI creates a window or panel from a View, a UIInfo object is
 created to act as the Handler's reference to that window and to the objects
-whose :term:`trait attribute`\ s are displayed in it. Each entry in the View's
-context (see :ref:`the-view-context`) becomes an attribute of the UIInfo
-object. [12]_ For example, the UIInfo object created in
-:ref:`Example 7 <example-7-using-a-multi-object-view-with-a-context>`
+whose :term:`trait attribute`\ s are displayed in it.  This object holds a
+reference to the UI instance in it's **ui** trait, and whether the or not
```
Comment (mdickinson) [inline docs/source/traitsui_user_manual/handler.rst:49]: "it's" -> "its"

Comment (mdickinson) [reply docs/source/traitsui_user_manual/handler.rst:49]: there's also a spurious "the" between "whether" and "or".

BTW, sorry for not grouping the prior comments together in a review; I'll do that from this point onwards.

File: docs/source/traitsui_user_manual/handler.rst [hunk 9]
```diff
@@ -45,21 +45,121 @@ means of the UIInfo object.
...
+   destroyed:
+      Set to True when the UI has finished being destroyed.
+
+
+.. rubric:: Useful UI methods
+
++---------------------------+--------------------------------------------------+
+|Method                     |Purpose                                           |
++===========================+==================================================+
+|dispose(result, abort)     |Disposes of the UI.  This can be called to close a|
```
Comment (mdickinson) [inline docs/source/traitsui_user_manual/handler.rst:156]: nit: can we have spaces around the content in the table entries (as in the examples [here](https://docutils.sourceforge.io/docs/user/rst/quickref.html#tables))? I don't think it makes a difference to the rendered result, but it makes it more readable in the source.

File: docs/source/traitsui_user_manual/index.rst [hunk 10]
```diff
@@ -21,3 +28,41 @@ TraitsUI |version| User Manual
...
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+THIS DOCUMENT, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+All trademarks and registered trademarks are the property of their respective
+owners.
+
+| Enthought, Inc.
+| 200 W Cesar Chavez, Suite 202
```
Comment (mdickinson) [inline docs/source/traitsui_user_manual/index.rst:63]: Is there value in including a postal address here?

File: docs/source/tutorials/code_snippets/event_loop_wx.py [hunk 11]
```diff
@@ -0,0 +1,10 @@
+from traits.api import HasTraits, Int
+import wx
```
Comment (mdickinson) [inline docs/source/tutorials/code_snippets/event_loop_wx.py:2]: Should we follow the usual import order and put 3rd party imports before ETS imports?

**Comment by mdickinson** [comment]
Closing and re-opening in the hope of getting a clean Travis CI run. If it still fails, I'll investigate further.

File: docs/source/traitsui_user_manual/factories_basic.rst (hunk 5 above)
Comment (corranwebster) [reply docs/source/traitsui_user_manual/factories_basic.rst:908]: No, that's a regression from previous clean-up.  Not sure why that's happening.

File: docs/source/traitsui_user_manual/handler.rst (hunk 9 above)
Comment (corranwebster) [reply docs/source/traitsui_user_manual/handler.rst:156]: I was following the style of other tables.  Can fix all.

Comment (mdickinson) [reply docs/source/traitsui_user_manual/handler.rst:156]: Up to you; nitpick level, not important.

Comment (corranwebster) [reply docs/source/traitsui_user_manual/handler.rst:156]: I think I am not going to do it in this PR.  There's probably a more thorough overhaul of documentation needed, but not today.
